### PR TITLE
Add Guillotina

### PR DESCRIPTION
**Pull request by masipcat**
**Checklist**

- [x] This project is explicitly related to ASGI.
- [x] The new list entry contains a project name, URL and description.
- [x] Alphabetical order is preserved across entries and sections.

**What is this project?**

 - [Guillotina](https://github.com/plone/guillotina) - Guillotina is the only full-featured Python AsyncIO REST Resource Application Server designed for high-performance, horizontally scaling solutions. 

**Do you know about other similar projects?**

No

**If so, how is this one different?**

---

_Anyone who agrees with this pull request can add a 👍._

### CODE REVIEW

**Comment by florimondmanca** [comment]
Heya,

Woah, that's a lot of :+1: already. 😄

Not being familiar with this project myself, can you point me to the docs page that mentions this project runs on ASGI? I was not able to grep through the docs (the search page doesn't load up). If it's not actually compatible with ASGI, it may be a better entry for `awesome-asyncio`.

Cheers

**Comment by masipcat** [comment]
the documentation hosted in readthedocs still points to 5.x version (that was based on aiohttp), but you can see here that 6.x implements ASGI: https://github.com/plone/guillotina/blob/618c6707c65520536753d14f9c22157659fd6b88/docs/source/developer/advanced.md#advanced

**Comment by masipcat** [comment]
Documentation updated! You can see in https://guillotina.readthedocs.io/en/latest/#build-a-guillotina-app that guillotina uses uvicorn

File: README.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ _Frameworks for building ASGI web applications._
 - [Channels](https://channels.readthedocs.io/en/latest/) - Asynchronous support for Django, and the original driving force behind the ASGI project. Supports HTTP and WebSockets with Django integration, and any protocol with ASGI-native code.
 - [Django](https://docs.djangoproject.com/en/3.0/topics/async/) - The web framework for perfectionists with deadlines. Has native ASGI support since version 3.0.
 - [FastAPI](https://github.com/tiangolo/fastapi) - A modern, high-performance web framework for building APIs with Python 3.6+ based on standard Python type hints. Powered by Starlette and Pydantic. Supports HTTP and WebSockets.
+- [Guillotina](https://github.com/plone/guillotina) - Guillotina is the only full-featured Python AsyncIO REST Resource Application Server designed for high-performance, horizontally scaling solutions.
```
Comment (florimondmanca) [inline README.md:42]: Suggestion to tune boilerplate down a bit and clarify how this relates to ASGI:

> Full-featured ASGI-compatible REST application framework, designed for high performance and horizontally scaling solutions.

Comment (masipcat) [reply README.md]: done :)

**Pull request merged**